src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService.cs

```diff
@@ -300,7 +303,7 @@ public Task<Document> AddEventAsync(Solution solution, INamedTypeSymbol destinat
             return GetEditAsync(
                 solution,
                 destination,
-                (t, opts, ai) => AddEvent(t, @event, opts, ai),
+                (t, opts, ai, _) => AddEvent(t, @event, opts, ai),
```

Comment:
When MAS generates a type or namespace, it iterates through all the members, generating them (and their nested types/namespaces). Generating the individual members is really fast, so I check the CancellationToken in the loop in AddMembers().

Generating the syntax an Event doesn't generate arbitrary extra syntax, so I didn't feel it would add value propagate the tokens there. (Cancelling MAS is already imperceptibly fast with just the tokens I have now).